loader/0002-quilt.mod.json.md

```diff
@@ -0,0 +1,362 @@
+
+### The `repositories` field
+| Type   | Required |
+|--------|----------|
+| Array  | False    |
+
+A list of repository url strings where dependencies can be looked for in addition to Quilt's central maven repository.
+
+### The `metadata` field 
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+Optional metadata that can be used by mods to display information about the mods installed.
+
+#### The `name` field
+| Type   | Required |
+|--------|----------|
+| String | False    |
+
+A human-readable name for this mod.
+
+#### The `description` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+A human-readable description of this mod. This description should be plain text, with the exception of line breaks, which can be represented with the newline character `\n`.
+
+#### The `contributors` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+A collection of `key: value` pairs denoting the persons or organizations that contributed to this project. The key should be the name of the person or organization, while the value can be either a string representing a single role or an array of strings each one representing a single role.
+
+A role can be any valid string. The "Owner" role is defined as being the person(s) or organization in charge of the project.
+
+#### The `contact` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+A collection of `key: value` pairs denoting various contact information for the people behind this mod, with all values being strings. The following keys are officially defined, though mods can provide as many additional values as they wish:
+* email — Valid e-mail address for the organization/developers
+* homepage — Valid HTTP/HTTPS address for the project or the organization/developers behind it
+* issues — Valid HTTP/HTTPS address for the project issue tracker
+* sources — Valid HTTP/HTTPS address for a source code repository
+
+#### The `license` field
+| Type                | Required |
+|---------------------|----------|
+| Array/String/Object | False    |
+
+The license or array of licenses this project operates under.
+
+A license is defined as either an [SPDX identifier](https://spdx.org/licenses/) string or an object in the following form:
+```json5
+{
+    name: "Perfectly Awesome License v1.0",
+    id: "PAL-1.0",
+    url: "https://theperfectlyawesomelicense.com/",
+    description: "This license does things and stuff and says that you can do things and stuff too!"
+}
+```
+The `"description"` field is optional.
+
+#### The `icon` field
+| Type          | Required |
+|---------------|----------|
+| Object/String | False    |
+
+One or more paths to a square .PNG file. If an object is provided, the keys must be the resolution of the corresponding file. For example:
```

Comment:
Is constraining the mod icon to only be a .PNG file necessary? [ImageIO on Java 8 natively supports at least 4 other image formats](https://docs.oracle.com/javase/8/docs/api/javax/imageio/package-summary.html) (more modern versions of Java support [an additional format](https://docs.oracle.com/en/java/javase/16/docs/api/java.desktop/javax/imageio/package-summary.html) on top of that), and has support for plugins which provide the ability to load other image formats. Additionally, would it be considered to be an error if a mod provides a non-square .PNG, or is handling this case left to whatever is parsing the quilt.mod.json file?

Reply:
The limitation to PNG is primarily because Minecraft doesn't super like binding textures for any other type of image, as far as I know.

Reply:
I would maybe be willing to support SVG, but not much else. A non-square PNG breaks spec, so that would be considered an error.

Reply:
SVGs would put quite the burden on those who implement this imo, even for stuff like Mod Menu.

Reply:
> SVGs would put quite the burden on those who implement this imo, even/especially for stuff like Mod Menu.

It certainly would, it's just the only other format I could justify including. PNG alone is fine.

Reply:
I would leave it PNG only. There are a lot of mod icons that are SVG, but any SVG icon can be rendered to a high resolution PNG image.

Reply:
The advantage to allowing SVG icons would be so that mods can have high resolution icons in-game without shipping large, high quality PNG's. I agree that it is probably best to forgo this, however.

Reply:
PNG only I would say.